pkg/media/audiotrack.go

```diff
@@ -257,10 +264,12 @@ func NewPCMRemoteTrack(track *webrtc.TrackRemote, writer *media.WriteCloser[medi
 	// resampledPCMWriter resamples the PCM16 samples from DefaultOpusSampleRate to targetSampleRate and
 	// writes them to the writer. If no resampling is needed, we directly point resampledPCMWriter to writer.
 	var isResampled bool
-	resampledPCMWriter := *writer
+	var resampledPCMWriter media.WriteCloser[media.PCM16Sample]
 	if targetSampleRate != DefaultOpusSampleRate {
-		resampledPCMWriter = media.ResampleWriter(*writer, targetSampleRate)
```

Comment:
ResampleWriter takes source sample rate as the parameter, not the target. It figures out the target using the SampleRate() method on the passed writer.